src/aleph/vm/conf.py

```diff
@@ -356,6 +357,14 @@ def setup(self):
 
         os.makedirs(self.EXECUTION_ROOT, exist_ok=True)
 
+        # If the Linux kernel provided is on another device than the execution root,
+        # copy it to the execution root to allow hardlink creation within jailer directories.
+        if os.stat(self.LINUX_PATH).st_dev != os.stat(self.EXECUTION_ROOT).st_dev:
+            logger.info("The Linux kernel is on another device than the execution root. Creating a copy.")
+            linux_path_on_device = self.EXECUTION_ROOT / "vmlinux.bin"
+            shutil.copy(self.LINUX_PATH, linux_path_on_device)
```

Comment:
Here also adding a `self.LINUX_PATH.isfile()` check, should be good to avoid not existing errors on copy

Reply:
`shutil.copy()` should override the existing file. We could optimize the process to only override it if the two files differ :thinking: